### ISSUE TITLE
loopover_devices :- can't goto PM_EQUIPTMENT_CONTROL menu

### ISSUE THREAD

**Issue by ballle98** [ballle98/AqualinkD#87]
```
12:57:44.282 Error:   PDA:       PDA Device programmer couldn't select menu item 'EQUIPMENT ON/OFF' menu 2
12:57:44.282 Error:   PDA:       PDA Device programmer request for menu 19, stuck on 2
12:57:44.282 Error:   PDA:       PDA Device programmer didn't find a requested menu 19, current 2
12:57:44.282 Error:   PDA:       loopover_devices :- can't goto PM_EQUIPTMENT_CONTROL menu
12:57:44.282 Error:   PDA:       PDA Init :- can't find menu
12:58:42.871 Error:   PDA:       PDA Device programmer couldn't select menu item 'EQUIPMENT ON/OFF' menu 2
12:58:42.871 Error:   PDA:       PDA Device programmer request for menu 19, stuck on 2
12:58:42.871 Error:   PDA:       PDA Device programmer didn't find a requested menu 19, current 2
12:58:42.871 Error:   PDA:       loopover_devices :- can't goto PM_EQUIPTMENT_CONTROL menu
12:58:42.871 Error:   PDA:       PDA Wake Init :- can't find menu
```

pi@raspberrypi:~/git/aquaweb $ journalctl -f -u aqualinkd
-- Journal begins at Wed 2023-02-22 08:50:07 CST. --
Nov 01 16:56:58 raspberrypi aqualinkd[8809]: PDA:       loopover_devices :- can't goto PM_EQUIPTMENT_CONTROL menu
Nov 01 16:56:58 raspberrypi aqualinkd[8809]: PDA:       PDA Init :- can't find menu
Nov 01 16:57:04 raspberrypi aqualinkd[8809]: PDA:       PDA Device programmer couldn't select menu item 'SET TEMP' menu 4
Nov 01 16:57:04 raspberrypi aqualinkd[8809]: PDA:       PDA Device programmer request for menu 7, stuck on 4
Nov 01 16:57:04 raspberrypi aqualinkd[8809]: PDA:       PDA Device programmer didn't find a requested menu 7, current 4
Nov 01 16:57:04 raspberrypi aqualinkd[8809]: PDA:       Could not get heater setpoints, trying again!
Nov 01 16:57:06 raspberrypi aqualinkd[8809]: PDA:       waitForPDAnextMenu - received STATUS instead of CLEAR
Nov 01 16:57:06 raspberrypi aqualinkd[8809]: PDA:       PDA Device programmer request for menu 15, stuck on 7
Nov 01 16:57:06 raspberrypi aqualinkd[8809]: PDA:       PDA Device programmer didn't find a requested menu 15, current 7
Nov 01 16:57:06 raspberrypi aqualinkd[8809]: PDA:       PDA Init :- Error getting freeze setpoints
Nov 01 16:57:43 raspberrypi aqualinkd[8809]: PDA:       waitForPDAnextMenu - received STATUS instead of HIGHLIGHT
Nov 01 16:57:50 raspberrypi aqualinkd[8809]: PDA:       loopover_devices :- can't find ALL OFF
Nov 01 16:57:50 raspberrypi aqualinkd[8809]: PDA:       PDA Wake Init :- can't find menu
Nov 01 16:59:39 raspberrypi aqualinkd[8809]: PDA:       waitForPDAnextMenu - received STATUS instead of CLEAR
Nov 01 16:59:39 raspberrypi aqualinkd[8809]: PDA:       PDA Device programmer request for menu 19, stuck on 2
Nov 01 16:59:39 raspberrypi aqualinkd[8809]: PDA:       PDA Device programmer didn't find a requested menu 19, current 2
Nov 01 16:59:39 raspberrypi aqualinkd[8809]: PDA:       loopover_devices :- can't goto PM_EQUIPTMENT_CONTROL menu
Nov 01 16:59:39 raspberrypi aqualinkd[8809]: PDA:       PDA Wake Init :- can't find menu
Nov 01 17:02:32 raspberrypi aqualinkd[8809]: PDA:       waitForPDAnextMenu - received STATUS instead of CLEAR
Nov 01 17:02:32 raspberrypi aqualinkd[8809]: PDA:       goto_pda_menu went from FW_VERSION to 1
Nov 01 17:03:15 raspberrypi aqualinkd[8809]: PDA:       waitForPDAnextMenu - received STATUS instead of HIGHLIGHT
Nov 01 17:03:21 raspberrypi aqualinkd[8809]: PDA:       waitForPDAMessageTypesOrMenu 0x08,0x04,0xff,(null),0 - Connection timed out
Nov 01 17:03:21 raspberrypi aqualinkd[8809]: PDA:       waitForPDAMessageTypesOrMenu: did not receive 0x08, 0x04 or 0xff
Nov 01 17:03:26 raspberrypi aqualinkd[8809]: PDA:       waitForPDAMessageTypesOrMenu 0x08,0x04,0xff,(null),0 - Connection timed out
Nov 01 17:03:26 raspberrypi aqualinkd[8809]: PDA:       waitForPDAMessageTypesOrMenu: did not receive 0x08, 0x04 or 0xff
Nov 01 17:03:26 raspberrypi aqualinkd[8809]: PDA:       loopover_devices :- can't find ALL OFF
Nov 01 17:03:26 raspberrypi aqualinkd[8809]: PDA:       PDA Wake Init :- can't find menu
Nov 01 17:04:38 raspberrypi aqualinkd[8809]: PDA:       PDA Device programmer couldn't select menu item 'EQUIPMENT ON/OFF' menu 2
Nov 01 17:04:38 raspberrypi aqualinkd[8809]: PDA:       PDA Device programmer request for menu 19, stuck on 2
Nov 01 17:04:38 raspberrypi aqualinkd[8809]: PDA:       PDA Device programmer didn't find a requested menu 19, current 2
Nov 01 17:04:38 raspberrypi aqualinkd[8809]: PDA:       loopover_devices :- can't goto PM_EQUIPTMENT_CONTROL menu
Nov 01 17:04:38 raspberrypi aqualinkd[8809]: PDA:       PDA Wake Init :- can't find menu
Nov 01 17:06:23 raspberrypi aqualinkd[8809]: PDA:       waitForPDAnextMenu - received STATUS instead of CLEAR
Nov 01 17:06:23 raspberrypi aqualinkd[8809]: PDA:       PDA Device programmer request for menu 19, stuck on 2
Nov 01 17:06:23 raspberrypi aqualinkd[8809]: PDA:       PDA Device programmer didn't find a requested menu 19, current 2
Nov 01 17:06:23 raspberrypi aqualinkd[8809]: PDA:       loopover_devices :- can't goto PM_EQUIPTMENT_CONTROL menu
Nov 01 17:06:23 raspberrypi aqualinkd[8809]: PDA:       PDA Wake Init :- can't find menu

**Comment by ballle98** [comment]
problem seems to go away when I turn on logging.  I think the send is coming too quickly after the receive.  added debug prints for timing.

```
11:04:30.737 Error:   RS Serial: Time from recv to blocking send is 0.000014944 sec
11:04:31.917 Error:   RS Serial: Time from recv to blocking send is 0.000031611 sec
11:04:31.944 Error:   RS Serial: Time from recv to blocking send is 0.000022981 sec
11:04:32.272 Error:   RS Serial: Time from recv to blocking send is 0.000025259 sec
11:04:32.607 Warning: PDA:       waitForPDAnextMenu - received STATUS instead of CLEAR
11:04:32.607 Error:   PDA:       PDA Device programmer request for menu 19, stuck on 2
11:04:32.607 Error:   PDA:       PDA Device programmer didn't find a requested menu 19, current 2
11:04:32.607 Error:   PDA:       loopover_devices :- can't goto PM_EQUIPTMENT_CONTROL menu
11:04:32.607 Error:   PDA:       PDA Init :- can't find menu
```

**Comment by ballle98** [comment]
No spec for aqualink protocol but looking at similar RS-485 protocol https://www.tascam.eu/en/docs/MX-8A_RS-485_protocol.pdf there is a min frame to frame time of 4 milliseconds and there seems to be issues with RPi 4 sending a response too quickly

![image](https://github.com/ballle98/AqualinkD/assets/2252524/59586ae0-33e1-4e77-8b26-0677a7b96c25)

**Comment by ballle98** [comment]
The fix for this (commit `fb96032`) only added 4 comment lines to `aq_serial.c` explaining the rationale for the frame delay logic. The actual frame delay code (`frame_delay`, `min_frame_wait_time`, `nanosleep`) was already present before this commit.

The same frame delay logic has been merged upstream via [aqualinkd/AqualinkD#260](https://github.com/aqualinkd/AqualinkD/pull/260) (without the comments). **Nothing of substance remains in this commit that is not already upstream.** This can be safely dropped during the rebase planned in #96.
